master/buildbot/config.py

```diff
@@ -816,6 +816,34 @@ def __init__(self, name=None, slavename=None, startSlavenames=None, slavenames=N
 
         self.description = description
 
+        self.customBuildUrls = customBuildUrls or {}
+        self._validateCustomBuildUrls()
+
+    def _validateCustomBuildUrls(self):
+        """
+        Validates that customBuildUrls is a dictionary containing only strings and in the format {'name': 'url'}
+        """
+        checkDictionaryContainsOnlyString = all(isinstance(key, str) and isinstance(value, str)
+                                                for key, value in self.customBuildUrls.iteritems())
+
+        if not isinstance(self.customBuildUrls, dict) or (
```

Comment:
if `self.customBuildUrls` is not a dict you would already get an error on the line above.